File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, Field


class NotificationRead(BaseModel):
    id: int
    title: str
    body: str
    notification_type: str
    channel: str
    status: str
    recipient_type: str
    recipient_id: int
    action_label: str | None = None
    action_path: str | None = None
    fail_reason: str | None = None
    sent_at: datetime


class NotificationUnreadCount(BaseModel):
    unread_count: int = Field(ge=0)

File: frontend/src/types/notification.ts
import type { PageParams } from './common'

export interface NotificationItem {
  id: number
  title: string
  body: string
  notification_type: 'todo' | 'notification' | 'announcement'
  channel: string
  status: 'unread' | 'read' | 'sent' | 'failed'
  recipient_type: string
  recipient_id: number
  action_label: string | null
  action_path: string | null
  fail_reason: string | null
  sent_at: string
}

export interface NotificationListParams extends PageParams {
  type?: 'todo' | 'notification' | 'announcement'
}

export interface NotificationUnreadCount {
  unread_count: number
}
